### ISSUE TITLE
Ensure we're using text compression/serving gzipped assets

### ISSUE THREAD

**Issue by joaoviana** [lightdash/lightdash#7097]
### Description

Currently, we don't serve gzipped assets on deployments. 
From a Lighthouse report, we can see that `echarts-x` JS asset is 2.5Mb: 
<img width="715" alt="image" src="https://github.com/lightdash/lightdash/assets/7611706/cef0f143-8dc3-4e63-b830-7ad773fc679a">

When in fact, when building the prod assets, we can see that the gzipped version is way smaller: 
<img width="978" alt="current-build" src="https://github.com/lightdash/lightdash/assets/7611706/c6591f18-3d85-47bc-b76a-81a6e9776c4f">

For reference, as well, on the network tab:
<img width="880" alt="image" src="https://github.com/lightdash/lightdash/assets/7611706/5e76662c-f412-4ee0-9379-0b9044f95955">

The assets are cached, so this only happens on first-time-load, but it could definitely be improved, since every time we release, we generate a new hash suffix, which will then invalidate the cache, and cause a re-fetch of the assets. 

### What problem does this solve?

FCP times. 

### Performance insights for echarts asset: 
<img width="261" alt="image" src="https://github.com/lightdash/lightdash/assets/7611706/cfb05b53-bde1-48ee-af3a-18ee17781494">
<img width="1203" alt="image" src="https://github.com/lightdash/lightdash/assets/7611706/de55d3b9-ef25-4d4f-b369-6329e04e568b">

**Comment by joaoviana** [comment]
tldr on gzip:
Render envs **compresses** the assets
Vitejs does **not** compress the assets
Current GCP config does **not** compress the assets because of the `Via` header and some other config we're missing.

**Comment by joaoviana** [comment]
Closed related PR as it didn't work. 
Alternative: do it on the express server side.

**Comment by joaoviana** [comment]
Sentry: https://lightdash.sentry.io/issues/4338486790/?project=5959292&query=&referrer=performance-related-issues-issue-stream&statsPeriod=24h
